aptos-indexer-processor-example/src/processors/events/events_version_tracker_savers.rs

```diff
@@ -0,0 +1,58 @@
+use crate::{
+    db::common::models::processor_status::ProcessorStatus,
+    schema::processor_status,
+    utils::database::{execute_with_better_error, ArcDbPool},
+};
+use anyhow::Result;
+use aptos_indexer_processor_sdk::common_steps::ProcessorStatusSaver;
+use aptos_indexer_processor_sdk::{
+    types::transaction_context::TransactionContext,
+    utils::{errors::ProcessorError, time::parse_timestamp},
+};
+use async_trait::async_trait;
+use diesel::{upsert::excluded, ExpressionMethods};
+pub struct ProcessorStatusSaverImpl {
```

Comment:
should we name this `DefaultProcessorStatusSaver`? also usually we leave out `Impl` from the struct names.

Reply:
I'm going with the latter suggestion as it prevents some duplicate code in the processor, too. That change will be in https://github.com/aptos-labs/aptos-indexer-processor-example/pull/6/files which has been rebased on top of this PR.